compile error on ubuntu with oce

When I build the latest branch v0.6.0 on linux with oce follow the guide 2b). use -DIFCXML_SUPPORT=off
 It fails in compilation stage as follows:

`/home/abay/code/IfcOpenShell/src/ifcgeom/IfcGeomShapes.cpp:1709:15: error: ‘Standard_Integer TCollection_BaseSequence::Size’ is protected within this context
   if (polygon.Size() < 3) {
               ^~~~
In file included from /usr/local/include/oce/TColgp_SequenceOfPnt.hxx:13:0,
                 from /home/abay/code/IfcOpenShell/src/ifcgeom/../ifcgeom/IfcGeom.h:47,
                 from /home/abay/code/IfcOpenShell/src/ifcgeom/IfcGeomShapes.cpp:107:
/usr/local/include/oce/TCollection_BaseSequence.hxx:130:20: note: declared protected here
   Standard_Integer Size;
                    ^~~~
/home/abay/code/IfcOpenShell/src/ifcgeom/IfcGeomShapes.cpp:1709:20: error: expression cannot be used as a function
   if (polygon.Size() < 3) {
                    ^
/home/abay/code/IfcOpenShell/src/ifcgeom/IfcGeomShapes.cpp: In member function ‘bool IfcGeom::KernelIfc4x3_rc1::convert(const Ifc4x3_rc1::IfcPolygonalFaceSet*, TopoDS_Shape&)’:
/home/abay/code/IfcOpenShell/src/ifcgeom/IfcGeomShapes.cpp:1828:15: error: ‘class TopTools_ListOfShape’ has no member named ‘Size’
     if (faces.Size() == 0) return false;
               ^~~~
/home/abay/code/IfcOpenShell/src/ifcgeom/IfcGeomShapes.cpp: In function ‘void {anonymous}::break_closed(const TopoDS_Wire&, std::vector<TopoDS_Wire>&)’:
/home/abay/code/IfcOpenShell/src/ifcgeom/IfcGeomShapes.cpp:1331:21: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   for (int i = 0; i < sorted_edges.size(); ++i) {
                   ~~^~~~~~~~~~~~~~~~~~~~~
/home/abay/code/IfcOpenShell/src/ifcgeom/IfcGeomShapes.cpp:1332:10: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
    if (i == sorted_edges.size() / 2) {
        ~~^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/abay/code/IfcOpenShell/src/ifcgeom/IfcGeomShapes.cpp:1326:10: warning: unused variable ‘u’ [-Wunused-variable]
   double u, v;
          ^
/home/abay/code/IfcOpenShell/src/ifcgeom/IfcGeomShapes.cpp:1326:13: warning: unused variable ‘v’ [-Wunused-variable]
   double u, v;
             ^
/home/abay/code/IfcOpenShell/src/ifcgeom/IfcGeomShapes.cpp: In function ‘bool {anonymous}::make_indexed_polygon(IfcGeom::KernelIfc4x1&, const std::vector<gp_Pnt>&, const std::vector<int>&, TopoDS_Wire&)’:
/home/abay/code/IfcOpenShell/src/ifcgeom/IfcGeomShapes.cpp:1709:15: error: ‘Standard_Integer TCollection_BaseSequence::Size’ is protected within this context
   if (polygon.Size() < 3) {
               ^~~~
In file included from /usr/local/include/oce/TColgp_SequenceOfPnt.hxx:13:0,
                 from /home/abay/code/IfcOpenShell/src/ifcgeom/../ifcgeom/IfcGeom.h:47,
                 from /home/abay/code/IfcOpenShell/src/ifcgeom/IfcGeomShapes.cpp:107:
/usr/local/include/oce/TCollection_BaseSequence.hxx:130:20: note: declared protected here
   Standard_Integer Size;
                    ^~~~
/home/abay/code/IfcOpenShell/src/ifcgeom/IfcGeomShapes.cpp:1709:20: error: expression cannot be used as a function
   if (polygon.Size() < 3) {
                    ^
/home/abay/code/IfcOpenShell/src/ifcgeom/IfcGeomShapes.cpp: In member function ‘bool IfcGeom::KernelIfc4x1::convert(const Ifc4x1::IfcPolygonalFaceSet*, TopoDS_Shape&)’:
/home/abay/code/IfcOpenShell/src/ifcgeom/IfcGeomShapes.cpp:1828:15: error: ‘class TopTools_ListOfShape’ has no member named ‘Size’
     if (faces.Size() == 0) return false;
               ^~~~
/home/abay/code/IfcOpenShell/src/ifcgeom/IfcGeomShapes.cpp: At global scope:`

How to fix this error? Thanks for any advice!

Similar problem here (line 1721 instead of 1709): 

`/usr/local/src/prg/IfcOpenShell/src/ifcgeom/IfcGeomShapes.cpp:1721:15: error: ‘Standard_Integer TCollection_BaseSequence::Size’ is protected within this context
   if (polygon.Size() < 3) {`

I'm using Ubuntu 18.04 and followed the installation procedure here: https://github.com/IfcOpenShell/IfcOpenShell#compiling-on-nix with the following cmake flags:

`cmake ../../IfcOpenShell/cmake/ -DOCC_LIBRARY_DIR=/usr/lib/x86_64-linux-gnu/ \
      -DOPENCOLLADA_INCLUDE_DIR="/usr/local/include/opencollada" \
      -DOPENCOLLADA_LIBRARY_DIR="/usr/local/lib/opencollada"  \
      -DPCRE_LIBRARY_DIR=/usr/lib/x86_64-linux-gnu/ \
      -DIFCXML_SUPPORT=Off`

Any advice?

Ca confirm that compilation worked also on my Ubuntu 18.04 setup with the fork of @lcpt. 
Nevertheless the compilation fails with output:
```
[ 99%] Linking CXX executable IfcConvert
libIfcParse.a(parse_ifcxml.cpp.o): In function `IfcParse::parse_ifcxml(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
parse_ifcxml.cpp:(.text+0x3ec): undefined reference to `xmlSAXUserParseFile'
collect2: error: ld returned 1 exit status
CMakeFiles/IfcConvert.dir/build.make:190: recipe for target 'IfcConvert' failed
make[2]: *** [IfcConvert] Error 1
CMakeFiles/Makefile2:79: recipe for target 'CMakeFiles/IfcConvert.dir/all' failed
make[1]: *** [CMakeFiles/IfcConvert.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[100%] Linking CXX shared module _ifcopenshell_wrapper.so
[100%] Built target _ifcopenshell_wrapper
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2

```

--- a/src/ifcgeom/IfcGeomShapes.cpp
+++ b/src/ifcgeom/IfcGeomShapes.cpp
@@ -1718,7 +1718,7 @@ namespace {
 		}
 		k.remove_duplicate_points_from_loop(polygon, true);
 
-		if (polygon.Size() < 3) {
+		if (polygon.Length() < 3) {
 			return false;
 		}
 
@@ -1837,7 +1837,7 @@ bool IfcGeom::Kernel::convert(const IfcSchema::IfcPolygonalFaceSet* pfs, TopoDS_
         }
     }
 
-    if (faces.Size() == 0) return false;
+    if (faces.IsEmpty() == 0) return false;
 
     return create_solid_from_faces(faces, shape);
 }

